optuna/_gp/acqf.py

```diff
@@ -123,16 +159,32 @@ def eval_acqf(acqf_params: AcquisitionFunctionParams, x: torch.Tensor) -> torch.
     )
 
     if acqf_params.acqf_type == AcquisitionFunctionType.LOG_EI:
-        return logei(mean=mean, var=var + acqf_params.acqf_stabilizing_noise, f0=acqf_params.max_Y)
+        # If there are no feasible trials, max_Y is set to -np.inf.
+        # If max_Y is set to -np.inf, we set logEI to zero to ignore it.
+        f_val = (
+            logei(mean=mean, var=var + acqf_params.acqf_stabilizing_noise, f0=acqf_params.max_Y)
+            if not np.isneginf(acqf_params.max_Y)
+            else 0.0
```

Comment:
Typing-wise, it would be better.
```suggestion
            else torch.tensor(0.0, dtype=torch.float64)
```